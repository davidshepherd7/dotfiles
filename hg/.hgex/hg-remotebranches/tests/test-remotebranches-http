#!/bin/sh

echo "[extensions]" >> $HGRCPATH
echo "graphlog=" >> $HGRCPATH
echo "remotebranches=$(echo $(dirname $(dirname $0)))/hg_remotebranches.py" >> $HGRCPATH

hg init test
cd test
echo foo>foo
mkdir foo.d foo.d/bAr.hg.d foo.d/baR.d.hg
echo foo>foo.d/foo
echo bar>foo.d/bAr.hg.d/BaR
echo bar>foo.d/baR.d.hg/bAR

hg commit -A -m 1
hg serve -p $HGPORT -d --pid-file=../hg.pid
cd ..
cat hg.pid >> $DAEMON_PIDS

echo % clone
hg clone http://localhost:$HGPORT/ copy 2>&1 | \
  sed -e 's/[0-9][0-9.]*/XXX/g' -e 's/[KM]\(B\/sec\)/X\1/'

cd copy
echo
echo % add a commit to the clone
echo alpha > alpha
hg add alpha
hg ci -m 'add alpha'

echo
echo % verify that the branchheads are stored properly
hg pull | sed "s/$HGPORT//"
hg log --graph
