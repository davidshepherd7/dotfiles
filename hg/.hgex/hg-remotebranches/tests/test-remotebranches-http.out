adding foo
adding foo.d/bAr.hg.d/BaR
adding foo.d/baR.d.hg/bAR
adding foo.d/foo
% clone
requesting all changes
adding changesets
adding manifests
adding file changes
added XXX changesets with XXX changes to XXX files
updating to branch default
XXX files updated, XXX files merged, XXX files removed, XXX files unresolved

% add a commit to the clone

% verify that the branchheads are stored properly
pulling from http://localhost:/
searching for changes
no changes found
@  changeset:   1:610fbbf9c9f6
|  tag:         tip
|  user:        test
|  date:        Thu Jan 01 00:00:00 1970 +0000
|  summary:     add alpha
|
o  changeset:   0:8b6053c928fe
   tag:         default/default
   user:        test
   date:        Thu Jan 01 00:00:00 1970 +0000
   summary:     1

